accepted/00000-use-dnf-for-rpm-repos.md

```diff
@@ -0,0 +1,177 @@
+
+The idea here would be the following:
+
+- The repository metadata downloading is delegated to DNF (which uses [librepo](http://rpm-software-management.github.io/librepo/) for the actual fetching)
+- The downloaded XML metadata can be manually read (packages, updates, products, etc).
+- Legacy metadata not normally handled by DNF could be fetched by [librepo](http://rpm-software-management.github.io/librepo/) or urlgrabber
+- Authentication with ULN repositories can be done by porting the YUM plugin to DNF
+
+#### How to calculate the dependencies?
+To access this information, we would use [the DNF API](https://dnf.readthedocs.io/en/latest/api.html) (OBS packages at [openSUSE:Factory](https://build.opensuse.org/package/show/openSUSE:Factory/dnf) - [openSUSE:Backports:SLE-15-SP1](https://build.opensuse.org/package/show/openSUSE:Backports:SLE-15-SP1/dnf)).
+
+#### Requirements for this approach:
+- We would need to use `python3-dnf` library ([OBS link](https://build.opensuse.org/package/show/openSUSE:Backports:SLE-15-SP1/dnf)) to do this.
+- For fetching metadata not referenced by `repomd.xml`, we could use the `python3-librepo` library ([OBS link](https://build.opensuse.org/package/show/openSUSE:Backports:SLE-15-SP1/librepo)).
+  + Alternatively, the `python3-urlgrabber` library ([OBS link](https://build.opensuse.org/package/show/openSUSE:Factory/python-urlgrabber)) could be used.
+
+#### DNF Python code examples
+- Some DNF use cases from Python: https://dnf.readthedocs.io/en/latest/use_cases.html
+- Working with DNF: http://abregman.com/2016/11/29/python-working-with-rpms/
+- DNF `reposync` subcommand code: https://github.com/rpm-software-management/dnf-plugins-core/blob/master/plugins/reposync.py
+  + This implementation will download all metadata referenced by `repomd.xml`, even stuff not currently handled by DNF itself.
+- Example of resolving dependencies in a repo with DNF: https://pagure.io/releng/blob/master/f/scripts/spam-o-matic
+- Some Python examples with librepo: https://github.com/rpm-software-management/librepo/tree/master/examples/python
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main drawback of this would be imposing the DNF stack as a dependency for the server.
+This is only slightly an issue in SUSE Linux Enterprise, as it's maintained in PackageHub.
+For the downstream SUSE Manager product, it is likely that the DNF stack packages would need
+to be imported to be maintained exclusively for being used with SUSE Manager.
+
+However, this is unlikely to be much of a maintenance burden as the community packaging
+is actively maintained and well-structured. The packager (who is also the author of this RFC)
+is responsive to Uyuni/SUSE Manager needs and the upstream DNF developers are interested
+and responsive to community consumers' needs.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Revert back to using YUM
+
+One alternative to resolve this issue is to roll back to using YUM. However, this is a bad idea.
```

Comment:
This is indeed something we can't do! Everything > openSUSE/SLE15 is Python3, so we can't revert to a Python2 solution.

Reply:
Of course, switching back to yum makes no sense here.